File: FilelistCreator/src/main/java/com/rkarp/flc/settings/AppConfiguration.java
package com.rkarp.flc.settings;

import net.rickiekarp.core.settings.LoadSave;
import com.rkarp.flc.model.Directorylist;
import com.rkarp.flc.model.Filelist;
import javafx.collections.FXCollections;
import javafx.collections.ObservableList;

public class AppConfiguration {

    /** Observable lists **/
    public static ObservableList<Filelist> fileData = FXCollections.observableArrayList();
    public static ObservableList<Directorylist> dirData = FXCollections.observableArrayList();
    public static ObservableList<String> unitList = FXCollections.observableArrayList("B", "KB", "MB", "GB", "TB");

    /** settings **/
    @LoadSave
    public static boolean subFolderCheck = true;

}

File: projects/JavaFXApps/FilelistCreator/src/main/java/net/rickiekarp/flc/settings/AppConfiguration.kt
package net.rickiekarp.flc.settings

import net.rickiekarp.core.settings.LoadSave
import net.rickiekarp.flc.model.Directorylist
import net.rickiekarp.flc.model.Filelist
import javafx.collections.FXCollections
import javafx.collections.ObservableList

object AppConfiguration {

    /** Observable lists  */
    var fileData = FXCollections.observableArrayList<Filelist>()
    var dirData = FXCollections.observableArrayList<Directorylist>()
    var unitList = FXCollections.observableArrayList("B", "KB", "MB", "GB", "TB")

    /** settings  */
    @LoadSave
    var subFolderCheck = true

}
